library/lib.sh

```diff
@@ -255,45 +255,45 @@ lsrConvertToCollection() {
 
 lsrGetManagedNodes() {
     local guests_yml=$1
-    sed --quiet --regexp-extended 's/(^managed.*):/\1/p' "$guests_yml" | sort
+    sed --quiet --regexp-extended 's/(^managed.*):$/\1/p' "$guests_yml" | sort
 }
 
 lsrGetNodes() {
     local guests_yml=$1
-    sed --quiet --regexp-extended 's/(^[^ ]*):/\1/p' "$guests_yml" | sort
+    sed --quiet --regexp-extended 's/(^[^ ]*):$/\1/p' "$guests_yml" | sort
 }
 
 lsrGetNodeName() {
     local guests_yml=$1
     local node_pat=$2
-    sed --quiet --regexp-extended "s/(^$node_pat.*):/\1/p" "$guests_yml"
+    sed --quiet --regexp-extended "s/(^$node_pat.*):$/\1/p" "$guests_yml"
 }
 
 lsrGetCurrNodeHostname() {
     local guests_yml=$1
     local ip_addr
     ip_addr=$(hostname -I | awk '{print $1}')
-    grep "primary-address: $ip_addr" "$guests_yml" -B 10 | sed --quiet --regexp-extended 's/(^[^ ]*):/\1/p'
+    grep "primary-address: $ip_addr" "$guests_yml" -B 10 | sed --quiet --regexp-extended 's/(^[^ ]*):$/\1/p'
 }
 
 lsrGetNodeIp() {
     local guests_yml=$1
     local node=$2
     # awk '$1=$1' to remove extra spaces
-    sed --quiet "/$node\:/,/^[^ ]/p" "$guests_yml"  | sed --quiet --regexp-extended 's/primary-address: (.*)/\1/p' | awk '$1=$1'
+    sed --quiet "/$node:/,/^[^ ]/p" "$guests_yml"  | sed --quiet --regexp-extended 's/primary-address: (.*)/\1/p' | awk '$1=$1'
```

Comment:
```suggestion
    sed --quiet "/$node:$/,/^[^ ]/p" "$guests_yml"  | sed --quiet --regexp-extended 's/primary-address: (.*)/\1/p' | awk '$1=$1'
```
?

Reply:
it searches for all entries of the requested $node. E.g. for `managed-node1` in the following context, the first match is for `managed-node1:$`. Further this sed prints from mathced `managed-node1` until next line not starting with space, which is `control-node1:`
```
managed-node1:
    -OPTIONLESS-FIELDS: &id001
      - primary_address
      - topology_address
      - facts
    primary-address: 192.168.124.24
    topology-address: 192.168.124.24
    role: managed_node
    become: false
    facts:
        in-sync: true
        arch: x86_64
        distro: CentOS Stream 9
...
control-node1:
OPTIONLESS-FIELDS: &id001
      - primary_address
      - topology_address
      - facts
...
```

Reply:
ok - so should it be `/$node:/` or `/$node:$/`?